src/corelibs/analogio/test_analogio_dac.cpp

```diff
@@ -0,0 +1,195 @@
+/**
+ * @brief test_analogio_dac.cpp
+ *
+ * @details This test verifies the DAC functionality.
+ * It writes different values via analogWrite() on the DAC output pin (TEST_PIN_ANALOG_IO_DAC, i.e., P14.8)
+ * and then reads the voltage using the ADC from TEST_PIN_ANALOG_IO_DAC_INPUT.
+ * The tests assert that the ADC reading is within tolerance compared to the expected value computed by:
+ *    expected_value = (775 - 93) * written_value / 1023 + 93
+ * The maximum output voltage of the dac is 2.5V, but V_DD is 3.3V. The upper bound is therefore: (2.5 / 3.3) * 1023 = 775
+*  The minimum output voltage of the dac is 0.3V. Therefore, the lower bound is: (0.3 / 3.3) * 1023 = 93
+ */
+
+// test includes
+#include "test_common_includes.h"
+#include "test_config.h"
+
+// defines
+#define TOLERANCE_BELOW 0.065
+#define TOLERANCE_ABOVE 0.03
+
+/**
+ * @brief Validates if the actual ADC value is within the acceptable range of the expected value.
+ *
+ * @param expected_value The expected ADC value.
+ * @param actual_value The ADC value read by the system.
+ * @return true if the actual value is within the acceptable range, false otherwise.
+ */
+static bool validate_adc_raw_value(int expected_value, int actual_value) {
+    return (expected_value * (1 - TOLERANCE_BELOW)) < actual_value &&
+           (expected_value * (1 + TOLERANCE_ABOVE)) > actual_value;
+}
+
+/**
+ * @brief Suite setup function, runs before test suite execution begins.
+ */
+static void analogio_dac_suite_setup() {
+    // Configure the DAC output pin as an output
+    pinMode(TEST_PIN_ANALOG_IO_DAC, OUTPUT);
+}
+
+/**
+ * @brief Suite teardown function, runs after test suite execution is complete.
+ */
+static void analogio_dac_suite_teardown() {
+    // Additional teardown if needed.
+}
+
+// Define test group name
+TEST_GROUP(analogio_dac);
+
+/**
+ * @brief Setup method called by Unity before every test in this test group.
+ */
+static TEST_SETUP(analogio_dac) { }
+
+/**
+ * @brief Tear down method called by Unity after every test in this test group.
+ */
+static TEST_TEAR_DOWN(analogio_dac) { }
+
+/**
+ * @brief Test 1: Write full-scale (ADC_RESOLUTION) and read back.
+ */
+TEST_IFX(analogio_dac, test_dac_write_and_read_value_full)
+{
+    // Configure ADC and DAC resolution (10-bit)
+    analogReference(DEFAULT);
+    analogWriteResolution(10);
+    
+    int write_value = ADC_RESOLUTION;
+    // Write the full-scale value (1023) to the DAC output pin (P14.8)
+    analogWrite(TEST_PIN_ANALOG_IO_DAC, write_value);
+    
+    // Discard the first ADC reading to allow settling
+    analogRead(TEST_PIN_ANALOG_IO_DAC_INPUT);
+    delay(500);
+    
+    // Calculate expected ADC value using the following formula: 775 is highest resolution, 93 is lowest resolution --> needed a way to compare
+    int expected_value = (775 - 93) * write_value / 1023 + 93;
```

Comment:
making this expected value calculation a macro `calculate_expected_value(write_value, ...)` will help with:
+ Document all those magic numbers in one place (readability)
+ Hide those magic numbers in the tests (readability)
+ It will be easier to implement conditionally compiled for other boards or even cores, with different "magic" calculations (reusability, OPC principle.